src/backend/api/Fusion.Resources.Domain/Commands/Departments/AddDelegatedResourceOwner.cs

```diff
@@ -24,25 +20,24 @@ public AddDelegatedResourceOwner(string departmentId, Guid responsibleAzureUniqu
 
         public class Handler : IRequestHandler<AddDelegatedResourceOwner>
         {
-            private readonly ResourcesDbContext db;
+            const string ResourceOwnerRole = "Fusion.Resources.ResourceOwner";
+            private readonly IFusionRolesClient rolesClient;
 
-            public Handler(ResourcesDbContext db)
+            public Handler(IFusionRolesClient rolesClient)
             {
-                this.db = db;
+                this.rolesClient = rolesClient;
             }
+
             public async Task<Unit> Handle(AddDelegatedResourceOwner request, CancellationToken cancellationToken)
             {
-                var delegatedResourceOwner = new DbDepartmentResponsible
+                var roleAssignment = await rolesClient.AssignRoleAsync(request.ResponsibleAzureUniqueId, new RoleAssignment
                 {
-                    DateCreated = DateTime.UtcNow,
-                    DateFrom = request.DateFrom,
-                    DateTo = request.DateTo,
-                    DepartmentId = request.DepartmentId,
-                    ResponsibleAzureObjectId = request.ResponsibleAzureUniqueId,
-                };
-
-                db.DepartmentResponsibles.Add(delegatedResourceOwner);
-                await db.SaveChangesAsync(cancellationToken);
```

Comment:
Just asking..
This change is intended ? (to move from resource-db-table to roles service?
If this table is not to be used anymore, we should remove it.

Reply:
yes, the change is intended. We could remove the table, but it's still in use when checking access. Removing the table would require a very complex migration from the table to the roles service. I would think it's better to wait untill all delegations have expired and delete at a later stage.